src/lib/i18n.test.js

```diff
@@ -0,0 +1,52 @@
+/* global describe, it, expect, beforeAll */
+// @ts-check
+import i18n, { localesList } from '../i18n.js'
+import { getLanguage, getCurrentLanguage } from './i18n.js'
+import languages from './languages.json'
+import { readdir } from 'node:fs/promises'
+
+const testEachLanguage = (fn) => {
+  Object.keys(languages).forEach((lang) => fn(lang))
+}
+
+const allLanguages = (await readdir('./public/locales', { withFileTypes: true }))
+  .filter(dirent => dirent.isDirectory())
+  .map(dirent => dirent.name)
+
+describe('i18n', function () {
+  it('should have a languages.json entry for each folder', function () {
+    const namedLocales = localesList.map(({ locale }) => locale)
+    expect(namedLocales).toEqual(allLanguages)
+  })
+
+  describe('getLanguage', function () {
+    it('returns unknown when given non-truthy input', () => {
+      expect(getLanguage()).toBe('Unknown')
+      expect(getLanguage('')).toBe('Unknown')
+      expect(getLanguage(null)).toBe('Unknown')
+      expect(getLanguage(undefined)).toBe('Unknown')
+    })
```

Comment:
i don't know if we need this, but I didn't want to change it without knowing for sure.